src/walletSdk/interactive/index.ts

```diff
@@ -0,0 +1,97 @@
+import axios from "axios";
+
+import { Anchor } from "../anchor";
+import { AssetNotSupportedError, ServerRequestFailedError } from "../exception";
+
+type ExtraFields = {
+  [key: string]: string;
+};
+
+export class Interactive {
+  private homeDomain = "";
+  private anchor: Anchor = null;
+
+  constructor(homeDomain, anchor) {
+    this.homeDomain = homeDomain;
+    this.anchor = anchor;
+  }
+
+  async deposit(
+    accountAddress: string,
+    assetCode: string,
+    authToken: string,
+    extraFields?: ExtraFields,
+    fundsAccountAddress?: string
+  ) {
+    return await this.flow(
+      accountAddress,
+      assetCode,
+      authToken,
+      "deposit",
+      extraFields,
+      fundsAccountAddress
+    );
+  }
+
+  async withdraw(
+    accountAddress: string,
+    assetCode: string,
+    authToken: string,
+    extraFields?: ExtraFields,
+    fundsAccountAddress?: string
+  ) {
+    return await this.flow(
+      accountAddress,
+      assetCode,
+      authToken,
+      "withdraw",
+      extraFields,
+      fundsAccountAddress
+    );
+  }
+
+  async flow(
```

Comment:
👋 @quietbits @Ifropc the kotlin sdk has an `assetGet` param [here](https://github.com/stellar/kotlin-wallet-sdk/blob/324ccb6dbcac3c723f8164c6eca0eada6b5d0f58/wallet-sdk/src/main/kotlin/org/stellar/walletsdk/anchor/Interactive.kt#L127)

why was this needed? I didn't see it used in an example. Any reason we can't just add the code like below (line 72) that automatically throws an error if the asset is not included in the anchor's serviceInfo?

```
   if (!assets.includes(assetCode)) {
      throw new AssetNotSupportedError(type, assetCode);
    }
```

Reply:
Yes, it's more or less the same (with assetGet it's just a bit less code)